ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/worldstate/ExecutorServiceBuilder.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.worldstate;
+
+import java.util.concurrent.ExecutorService;
+
+public interface ExecutorServiceBuilder {
```

Comment:
A `Supplier<ExecutorService>` would be cleaner.  We don't need to do reflection or reification of the types.